app/models/package_manager/maven.rb

```diff
@@ -109,45 +110,58 @@ def self.mapping(raw_project, depth = 0)
       nil
     end
 
-    def self.mapping_from_pom_xml(version_xml, depth = 0)
-      xml = if version_xml.respond_to?("project")
-              version_xml.project
-            else
-              version_xml
-            end
-
-      parent = {
-        description: nil,
-        homepage: nil,
-        repository_url: "",
-        licenses: "",
-        properties: {},
-      }
-      if xml.locate("parent").present? && depth < MAX_DEPTH
-        group_id = extract_pom_value(xml, "parent/groupId")&.strip
-        artifact_id = extract_pom_value(xml, "parent/artifactId")&.strip
-        version = extract_pom_value(xml, "parent/version")&.strip
-        if group_id && artifact_id && version
-          parent = mapping_from_pom_xml(
-            get_pom(group_id, artifact_id, version),
-            depth + 1
-          )
+    def self.get_pom_documents(group_id, artifact_id, version, depth = 0)
+      return [] if depth >= MAX_DEPTH
+
+      pom_document = get_pom(group_id, artifact_id, version)
+      if pom_document.respond_to?("project")
```

Comment:
I would have thought this had to be a symbol `respond_to?(:project)` but if it works it works

Reply:
Apparently it's either. I see other `respond_to?("project")` elsewhere in this file though, so keeping it for consistency.